src/main/scala/scala/async/internal/AnfTransform.scala

```diff
@@ -259,6 +272,44 @@ private[async] trait AnfTransform {
         }
       }
 
+      // Replace the label parameters on `matchEnd` with use of a `matchRes` temporary variable
```

Comment:
maybe a bit of clarification here
  - the method name suggests that it eliminates multiple label parameters, but it expects there to be one
  - quickly recap the structure generated by patmat. is the following correct?
    - normal case defs are label defs without parameters.
    - the single matchEnd label def has excatly one (zero or one? it seems to have one also for unit-typed matches) parameter.